src/plugins/ui_actions/public/actions/action.ts

```diff
@@ -72,3 +75,48 @@ export interface Action<Context = {}, T = ActionType> {
    */
   execute(context: Context): Promise<void>;
 }
+
+/**
+ * A convenience interface used to register an action.
+ */
+export interface ActionDefinition<
```

Comment:
Now that this has been merged to master after my typescript PR, this is a duplicate type, there is a file `action_definition` with a `ActionDefinition` type as well.  I think we should use that one, and like @Dosant suggested, tie Config shape with the factory which I think should go in x-pack.